### ISSUE TITLE
[Feature request] Possibility to store file with name author - title or full name in details

### ISSUE THREAD

**Issue by update-freak** [mrmn2/PdfDing#47]
Hi,
I saw the PDF files are stored after uploading on the server in the folder data with a "ID"-name.
It's hard to find the correct pdf if I search for a specific file.
If its possible it would be good to have the possibility to name it to "author - title" on the server OR to see the full filename in details.

Edit: paperless ngx creates a folder with the name of the author/company and the document is named based on date and title.
I think a folder here is not necessary, but author - title would be helpful or the full filename in details.

Background: I managed my pdf files currently in folders and think of removing the whole structure and add them all to pdfding, but not sure if I should keep my documents also in the folders additonally, but I would like to did not have redundant data. Therefore I think I will prefer to remove my folder structure and add all to pdfding.

**Comment by mrmn2** [comment]
Thanks for the additional info. I can think of some already existing solutions that could help:

* You could use the author as a tag for the pdf. Then you can search for the title + author in the ui. Say you have a pdf named `some pdf` of author `some author` and tag the pdf with `some_author`. Then you could search for the pdf with `some pdf #some_author`.
* You could add the author + title to the description. (At some point in the feature I want to implement a feature for searching the description of pdfs.)

Would one of these solutions work for you?

If you are worried being locked in to PdfDing because of the random file names in the media folder I can think of the following solution: There could be a command the admin can execute in the shell of the running container that will copy all pdfs of an user to `media/export/<user_id>`. Inside this folders there will be sub directories named after the tags. The pdfs will be in these tag folders and the filenames will correspond to those in the ui. What do you think?

**Comment by update-freak** [comment]
Regarding the filtering / searching: I think I will do it like this:
- for books: I will name it as author(s) - title with the tag `#Buch` (book)
- for lecture scripts from university: I will name it as type of document (e.g. script, slides, tutorial, etc.) and the title of the lecture, e.g. Vorlesungsskript Elektrotechnik. The tag I will use for the university. And in the description I will mention the professor and maybe the year of the lecture.
--> so this naming structure works for me. (If you have ideas for improvement let me know ;) )

For the random file name, for me a export function is not needed (since I want to avoid to have a file double on my server).
I will upload every document in Pdfding and delete my current structure (but then the tracebility is important for me).
This could be important for the following usecase:

For example when I want to open a PDF file which another program to modify it, but its already uploaded to PDFding and I did not want to upload it again and fill out again the tags, description, etc. In that case it would be good to know the filename that I can find it directly to modify it.

![bild - Kopie](https://github.com/user-attachments/assets/258b9411-8b20-41f7-b76c-4fce81e248d0)

Possible solution. What do you think?
![example - Kopie](https://github.com/user-attachments/assets/1b0fea86-7db1-43c2-b9c5-44ebe9ac7d83)

**Comment by mrmn2** [comment]
Yes I think that your system is a good solution. You could also do multi-level tags, e.g `kit/kaufmann`. At the end it depends on what's most intuitive for you :)

Regarding the feature request I'll soon implement your suggestion with a minor adjustment. As the the string is the ID of the PDF file, I'll name it ID in the details page. You can expect it to be part of the next release.

**Comment by update-freak** [comment]
Hi @mrmn2,
I'm wondering why the ID (which is also used in the URL) is different from the filename?
![id - Kopie](https://github.com/user-attachments/assets/49c64003-998e-4c1d-8dc6-29869f0ed369)

**Comment by mrmn2** [comment]
ahhh, the file name is generated differently. I'll adjust it so that for new files the name is the same as the ID. I'll also fix that the ID shown in the details, so that it will point to the right file. Sorry for the inconvenience.

**Comment by update-freak** [comment]
This is then part of the next release (do you know when)? (I ask because I want to shift my documents soon ;) )

**Comment by mrmn2** [comment]
Yes it will be part of the next release, which should be ready this week. 

If you want to proceed faster you can also use the `dev_2025_01_10` tag, I created this tag to test something related to the demo mode with it. Don't rely on this tag though, as I will delete it with the next release.
